Xamarin.Forms.Platform.iOS/Renderers/MediaElementRenderer.cs

```
@@ -0,0 +1,475 @@
+					{
+						// used for a file embedded in the application package
+						asset = AVAsset.FromUrl(NSUrl.FromFilename(uriSource.Uri.LocalPath.Substring(1)));
+					}
+					else if (uriSource.Uri.Scheme == "ms-appdata")
+					{
+						string filePath = string.Empty;
+
+						if (uriSource.Uri.LocalPath.StartsWith("/local"))
+						{
+							filePath = Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.MyDocuments), uriSource.Uri.LocalPath.Substring(7));
+						}
+						else if (uriSource.Uri.LocalPath.StartsWith("/temp"))
+						{
+							filePath = Path.Combine(Path.GetTempPath(), uriSource.Uri.LocalPath.Substring(6));
+						}
+						else
+						{
+							throw new ArgumentException("Invalid Uri", "Source");
+						}
+
+						asset = AVAsset.FromUrl(NSUrl.FromFilename(filePath));
+					}
+					else
+					{
+						asset = AVUrlAsset.Create(NSUrl.FromString(uriSource.Uri.AbsoluteUri));
+					}
+				}
+				else
+				{
+					var fileSource = MediaElement.Source as FileMediaSource;
+					if (fileSource != null)
+					{
+						asset = AVAsset.FromUrl(NSUrl.FromFilename(fileSource.File));
+					}
+				}
+
+				
+				AVPlayerItem item = new AVPlayerItem(asset);
+				RemoveStatusObserver();
+
+				_statusObserver = (NSObject)item.AddObserver("status", NSKeyValueObservingOptions.New, ObserveStatus);
+				
+				
+				if (_avPlayerViewController.Player != null)
+				{
+					_avPlayerViewController.Player.ReplaceCurrentItemWithPlayerItem(item);
+				}
+				else
+				{
+					_avPlayerViewController.Player = new AVPlayer(item);
+					_rateObserver = (NSObject)_avPlayerViewController.Player.AddObserver("rate", NSKeyValueObservingOptions.New, ObserveRate);
+				}
+				
+				if (MediaElement.AutoPlay)
+				{
+					var audioSession = AVAudioSession.SharedInstance();
+					NSError err = audioSession.SetCategory(AVAudioSession.CategoryPlayback);
+					audioSession.SetMode(AVAudioSession.ModeMoviePlayback, out err);
+					err = audioSession.SetActive(true);
```

Possible errors are ignored. Perhaps the best solution would be to add an error event to the `MediaElement` class?